esp/src/src-react/components/Sasha.tsx

```diff
@@ -62,36 +120,125 @@ export const Sasha: React.FunctionComponent<SashaProps> = () => {
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="State"
+                value={stateFilter}
+                onChange={(event, newValue?: string) => setStateFilter(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="From Date"
+                value={fromDate}
+                onChange={(event, newValue?: string) => setFromDate(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="To Date"
+                value={toDate}
+                onChange={(event, newValue?: string) => setToDate(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="Before WU"
+                value={beforeWU}
+                onChange={(event, newValue?: string) => setBeforeWU(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="After WU"
+                value={afterWU}
+                onChange={(event, newValue?: string) => setAfterWU(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <TextField
+                label="Output Fields"
+                value={outputFields}
+                onChange={(event, newValue?: string) => setOutputFields(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+              <Stack horizontal tokens={{ childrenGap: 20 }}>
+                <Checkbox
+                  label="Archived"
+                  checked={archived}
+                  onChange={(e, checked) => setArchived(!!checked)}
+                />
+                <Checkbox
+                  label="Online"
+                  checked={online}
+                  onChange={(e, checked) => setOnline(!!checked)}
+                />
+                <Checkbox
+                  label="Include DT"
+                  checked={includeDT}
+                  onChange={(e, checked) => setIncludeDT(!!checked)}
+                />
+                <Checkbox
+                  label="Descending"
+                  checked={descending}
+                  onChange={(e, checked) => setDescending(!!checked)}
+                />
+              </Stack>
+            </Stack>
+          ) : (
+            (["restoreECLWorkUnit", "restoreDFUWorkUnit", "archiveECLWorkUnit", "archiveDFUWorkUnit"].includes(selectedOption)) && (
+              <TextField
+                label={nlsHPCC.WUID}
+                value={wuid}
+                onChange={(event, newValue?: string) => setWuid(newValue || "")}
+                styles={{ fieldGroup: { width: 400 } }}
+              />
+            )
+          )}
+
+          <PrimaryButton type="submit">{nlsHPCC.Submit}</PrimaryButton>
```

Comment:
And if you were to add some kind of margin or padding (like `... style={{ marginTop: 10 }}`) to this button, that'll make it not be completely flush to any fields above it